.travis.yml

```diff
@@ -19,20 +19,30 @@ notifications:
       - carl.r.delsey@intel.com
       - matthew.k.hansen@intel.com
 
+env:
+  - CMAKE_BUILD_TYPE=Release
+
 before_install:
-  - if [ "${TRAVIS_REPO_SLUG}" != "ros-planning/navigation2" ]; then echo "Travis CI is supported only in ros-planning/navigation2" && exit 1; fi
-  - docker build -t nav2:latest --build-arg PULLREQ=$TRAVIS_PULL_REQUEST .
+  - if [ "${TRAVIS_REPO_SLUG}" != "ros-planning/navigation2" ];
+      then echo "Travis CI is supported only in ros-planning/navigation2" && exit 1;
+    fi
+  - docker build --tag navigation2:latest
+      --build-arg PULLREQ=$TRAVIS_PULL_REQUEST
+      --build-arg CMAKE_BUILD_TYPE
+      ./
 
 script:
-  - docker run --name nav2_bash --rm -d -i -t --net=host nav2:latest bash -l
-  - docker exec nav2_bash bash -c '. /ros2_ws/navigation2_ws/install/setup.bash &&
-      colcon test --packages-skip nav2_system_tests'
-  - docker exec nav2_bash bash -c "colcon test-result --verbose"
-  - docker exec nav2_bash bash -c ". install/setup.bash && cd build/nav2_system_tests; ./ctest_retry.bash 3"
+  - docker run --rm --detach --name nav2_bash navigation2:latest
+      sleep infinity
+  - docker exec --interactive --tty nav2_bash /ros_entrypoint.sh
+      colcon test --packages-skip nav2_system_tests
+  - docker exec --interactive --tty nav2_bash /ros_entrypoint.sh
+      colcon test-result --verbose
```

Comment:
```suggestion
      colcon test-result --verbose
  - docker exec --interactive --tty nav2_bash /ros_entrypoint.sh
      cp src/navigation2/tools/ctest_retry.bash build/nav2_system_tests
  - docker exec --interactive --tty nav2_bash /ros_entrypoint.sh
       bash -c "cd build/nav2_system_tests ; ./ctest_retry.bash 3"
```

Reply:
This re-enables the system test without copying an extra file into the image. The file was already there.

Obviously, it would be better if the ctest_retry.bash script were rewritten to allow it to be exectuted in place, but I figure that can be left for a future change.

Reply:
No. There is no log. I was blind and didn't delete the comment when I deleted the log code from the original script. The original script used to echo some comments on each test run to a file in `/tmp`.

Here is the original script for reference.
```
#!/bin/bash

fail=0
total=$1
# Change the file path manually to where you want to log fails
file=/tmp/fails.txt
echo "Total Loops = " $total
for ((i=1;i<=total;i++))
  do
    ctest
    result=$?
    echo "RESULT =" $result
    if [ "$result" != "0" ]
    then
      ((fail+=1))
      echo "TEST $i FAILED" >> $file
    fi
    echo $i "LOOPS COMPLETED"
    echo $fail "TOTAL FAILURES"
  done
echo $fail " LOOP FAILURES / " $total
echo $fail " LOOP FAILURES / " $total >> $file
```